3.9.1

remyperona opened the pull request:
- Enhancement: Improve Google Fonts Optimization by preloading and asychronously loading the Google Fonts URL (#2772)
- Enhancement: Automatically exclude invalid script types from delay JS (#3944, #4158)
- Enhancement: Improve helper text for the delay JS exclusions field (#4044)
- Bugfix: Prevent high CPU usage on some hosts when using remove unused CSS (#4072)
- Bugfix: Update our minification library to fix several issues (#3698, #3398, #3279, #2974, #2083)
- 3rd party compatibility: Prevent display issue of the images gallery on a WooCommerce product page when delay JS is enabled (#4077)
- 3rd party compatibility: Add additional exclusions from combine/defer JS (#4009, #4012, #3984, #4048, #4068, #4128)

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_CA

translation completed for the source file '/languages/rocket.pot'
on the 'fr_CA' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in pt_PT

translation completed updated for the source file '/languages/rocket.pot'
on the 'pt_PT' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in pt_PT

translation completed updated for the source file '/languages/rocket.pot'
on the 'pt_PT' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in pt_PT

translation completed updated for the source file '/languages/rocket.pot'
on the 'pt_PT' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in hr

translation completed for the source file '/languages/rocket.pot'
on the 'hr' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in hr

translation completed updated for the source file '/languages/rocket.pot'
on the 'hr' language.

dependabot[bot] committed:
Bump postcss from 7.0.35 to 7.0.36

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

transifex-integration[bot] committed:
Translate /languages/rocket.pot in zh_CN

translation completed for the source file '/languages/rocket.pot'
on the 'zh_CN' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in zh_CN

translation completed updated for the source file '/languages/rocket.pot'
on the 'zh_CN' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in zh_CN

translation completed updated for the source file '/languages/rocket.pot'
on the 'zh_CN' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in zh_CN

translation completed updated for the source file '/languages/rocket.pot'
on the 'zh_CN' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in zh_CN

translation completed updated for the source file '/languages/rocket.pot'
on the 'zh_CN' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

gautam-jha committed:
Update README.md

transifex-integration[bot] committed:
Translate /languages/rocket.pot in tr_TR

translation completed updated for the source file '/languages/rocket.pot'
on the 'tr_TR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in fr_FR

translation completed updated for the source file '/languages/rocket.pot'
on the 'fr_FR' language.

mostafa-hisham committed:
Closes #3112 Remove clear cloudflare cache from admin bar (PR #3983)

mostafa-hisham committed:
Fixes #3944 Delay JS exclude scripts with invalid type (PR #4049)

* delay js exclude scripts with invalid type
* remove ld+json condition
* update pattern to remove lazy quantifiers
* update fixtures

Co-authored-by: Rémy Perona <remy@wp-rocket.me>

transifex-integration[bot] committed:
Translate /languages/rocket.pot in tr_TR

translation completed updated for the source file '/languages/rocket.pot'
on the 'tr_TR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in tr_TR

translation completed updated for the source file '/languages/rocket.pot'
on the 'tr_TR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in tr_TR

translation completed updated for the source file '/languages/rocket.pot'
on the 'tr_TR' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in tr_TR

translation completed updated for the source file '/languages/rocket.pot'
on the 'tr_TR' language.

remyperona committed:
remove unused images

remyperona committed:
Revert "remove unused images"

This reverts commit 1a807b258b9c8770196d9a375baf4bad60cbcbfa.

Commit:
Update Minify dependency with latest changes (PR #3907)

* update minify dependency with latest changes
* fix incorrect replacement in case of duplicated values
* fix pattern to account for spacing
* prevent rewritting path for clip path with svg
* prevent stripping JS content matched as comment incorrectly

Commit:
Closes #3997 Optimize images used in the settings page (PR #4087)

* delete no longer used images
* optimize images
* remove unused images

remyperona committed:
Update pull request template

NataliaDrause committed:
Add new inline JS exclusion to prevent the cache_dir_size issue (PR #4009)

vmanthos committed:
Exclude www.vbt.io/ext/vbtforms.js from JS minification (PR #4012)

DahmaniAdame committed:
Closes #3984 Add new external script exclusion from JS minification (PR #4039)

vmanthos committed:
Exclude WATI  inline JS to prevent the increase of the /cache/min size (PR #4048)

vmanthos committed:
Add additional inline exclusions from combine JS (PR #4068)

NataliaDrause committed:
Add new inline JS exclusion to prevent the cache_dir_size issue (PR #4128)

engahmeds3ed committed:
Delay JS - Add explanatory text about default exclusions (PR #4141)

* Add helper attribute to delay js exclusion list field
* fix review changes
* adjust default exclusion list styling
* show another helper message if exclusions list has the default one
* fix review comments
* make code color black

Commit:
Update delay JS script to latest version (PR #4158)

* update script to latest version
* update script in fixture

mostafa-hisham committed:
Closes #2772 Preload & load Google Fonts asynchronously (PR #4110)

* load google fonts async
* fix test case shouldCombineMultipleTagsWithMultipleFamiliesInTag
* add hide_comments() to the trait
* remove dependency on AbstractOptimization and new methods
- Add astract optimize() method
- Add get_optimized_markup() method
* Use RegexTrait, remove check for GF number and use new methods
* Update fixtures and tests
* skip styles with a media="print" attribute

Co-authored-by: Rémy Perona <remperona@gmail.com>
Co-authored-by: Ahmed Saeed <eng.ahmeds3ed@gmail.com>

engahmeds3ed committed:
Fixes #4072 Prevent high CPU load from resource fetcher (PR #4139)

engahmeds3ed committed:
Auto-exclusion for WooCommerce Product Gallery when Delay JS is enabled (PR #4142)

remyperona committed:
update version to 3.9.1

remyperona committed:
Merge branch 'develop' into transifex

remyperona committed:
update po file with new strings in 3.9.1

GeekPress commented:
For #4077, we forgot to add jQuery migrate in the exclusion list => `/jquery-migrate(.min)?.js`

engahmeds3ed commented:
https://make.wordpress.org/core/2020/06/29/updating-jquery-version-shipped-with-wordpress/
jquery migrate is removed starting from WordPress version 5.7 so we may detect by the version of WordPress to add this jquery migrate to the exclusions list.
I will create a separate PR for this one.

Question: will this PR https://github.com/wp-media/wp-rocket/issues/4044 be affected by this so we need to add also jquery migrate along with jquery?

GeekPress commented:
> Question: will this PR #4044 be affected by this so we need to add also jquery migrate along with jquery?

Yes indeed, we need to add jQuery migrate on that PR too :/

transifex-integration[bot] committed:
Translate /languages/rocket.pot in pt_PT

translation completed for the source file '/languages/rocket.pot'
on the 'pt_PT' language.

transifex-integration[bot] committed:
Translate /languages/rocket.pot in tr_TR

translation completed for the source file '/languages/rocket.pot'
on the 'tr_TR' language.

engahmeds3ed committed:
Add jquery migrate to delay JS exclusions for woocommerce gallery and default list (PR #4168)

* add jquery migrate to delay js exclusions for woocommerce gallery and default exclusions list
* fix tests
* update test for wp_version check

Co-authored-by: Rémy Perona <remperona@gmail.com>

transifex-integration[bot] committed:
Translate /languages/rocket.pot in de_DE

translation completed for the source file '/languages/rocket.pot'
on the 'de_DE' language.

iCaspar committed:
Add phpunit/php-file-iterator v2.0.3 to composer (#4181)

iCaspar committed:
Fix/admin css code tag regression (#4182)

* Add a pre tag around code for delayJS helper

* Modify CSS to format <code><pre> text

* Run gulp sass

* Remove background shading for code inside pre

remyperona committed:
update translations files

remyperona committed:
Merge branch 'transifex' into develop

AlbertCintas committed:
CICD info behind secrets

And remove actions not used anymore
